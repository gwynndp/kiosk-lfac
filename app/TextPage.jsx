import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import MainNav from './MainNav';
import Sidebar from './Sidebar';

const PageWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: stretch;
  min-width: 70%;
`;

const PageHeader = styled.header`
  background-color: rgba(255, 255, 255, 0.5);
  margin: 0 0 10px;
  display: flex;
  justify-content: center;
  text-align: center;
  font-size: 24px;
`;

const Content = styled.div`
  background-color: rgba(255, 255, 255, 0.5);
  display: flex;
  justify-content: flex-start;
  flex-wrap: wrap;
  flex-grow: 1;
  height: auto; //580px for Firefox
  padding: 30px;
  overflow: scroll;
`;

export default function TextPage({ page }) {
  return (
    <React.Fragment>
      <MainNav gallery='false' />
      <PageWrapper>
        <PageHeader>
          <h1>{page.title}</h1>
        </PageHeader>
        <Content>{page.content}</Content>
      </PageWrapper>
      <Sidebar />
    </React.Fragment>
  );
}

TextPage.propTypes = {
  page: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string])),
};
