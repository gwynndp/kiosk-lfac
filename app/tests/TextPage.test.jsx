import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import TextPage from '../TextPage';

test('should render a TextPage', () => {
  const props = {
    page: {
      title: "Valley of the Moon Children's Foundation",
      content:
        '<img src="images/vom-logo.png" alt="" width="200" height="164" />\\n<p>Judge Arnold Rosenfeld formed the Valley of the Moon Foundation in 1994.  It’s mission is “in partnership with the Valley of the Moon Children’s Home and the Redwood Children’s Center to positively impact the present and future lives of the abused, abandoned, and neglected children in Sonoma County and to assure the children we serve positive opportunities to aid them to become productive, equal members of the community”.</p>\\n',
    },
  };

  const tree = renderer
    .create(
      <Router>
        <TextPage page={props.page} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// should render a MainNav
// should render a PageHeader & Content
// should render a Sidebar
