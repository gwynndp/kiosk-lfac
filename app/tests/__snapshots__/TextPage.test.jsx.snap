// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a TextPage 1`] = `
Array [
  <div
    className="sc-AykKC duquAX"
  >
    <a
      className="sc-AykKD dWbVqE"
      exact="true"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="home"
        src="images/kiosk_home.png"
        width="100px"
      />
    </a>
  </div>,
  <div
    className="sc-AykKH eRVEGl"
  >
    <header
      className="sc-AykKI knYcmi"
    >
      <h1>
        Valley of the Moon Children's Foundation
      </h1>
    </header>
    <div
      className="sc-AykKJ ljPQPk"
    >
      &lt;img src="images/vom-logo.png" alt="" width="200" height="164" /&gt;\\n&lt;p&gt;Judge Arnold Rosenfeld formed the Valley of the Moon Foundation in 1994.  It’s mission is “in partnership with the Valley of the Moon Children’s Home and the Redwood Children’s Center to positively impact the present and future lives of the abused, abandoned, and neglected children in Sonoma County and to assure the children we serve positive opportunities to aid them to become productive, equal members of the community”.&lt;/p&gt;\\n
    </div>
  </div>,
  <div
    className="sc-AykKF cvNFcn"
  >
    <div
      id="logos-wrapper"
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": "0 10px 20px 0",
        }
      }
    >
      <img
        alt="Cloverdale City logo"
        src="images/cloverdale-city-logo.png"
        width="100%"
      />
      <img
        alt="Cloverdale High School logo"
        src="images/cloverdale-hs-logo.png"
        width="100%"
      />
      <img
        alt="Listening For a Change logo"
        src="images/lfac_logo_black.gif"
        width="100%"
      />
    </div>
    <a
      className="sc-AykKG esZqhL"
      href="/lfac"
      onClick={[Function]}
    >
      <span>
        Listening for a Change
      </span>
    </a>
    <a
      className="sc-AykKG esZqhL"
      href="/schools"
      onClick={[Function]}
    >
      <span>
        Schools
      </span>
    </a>
    <a
      className="sc-AykKG esZqhL"
      href="/fires"
      onClick={[Function]}
    >
      <span>
        Fire Stories
      </span>
    </a>
    <a
      className="sc-AykKG esZqhL"
      href="/jackdemayo"
      onClick={[Function]}
    >
      <span>
        Remembering Jack DeMayo
      </span>
    </a>
    <a
      className="sc-AykKG esZqhL"
      href="/vom"
      onClick={[Function]}
    >
      <span>
        Valley of the Moon
      </span>
    </a>
  </div>,
]
`;
